File: subprojects/docs/src/samples/templates/gradle-plugin-in-java/src/test/java/com/example/plugin/GreetingPluginTest.java
package com.example.plugin;

import org.gradle.testfixtures.ProjectBuilder;
import org.gradle.api.Project;
import org.junit.Test;
import static org.junit.Assert.assertNotNull;


public class GreetingPluginTest {
    @Test
    public void pluginRegistersATask() {
        // Create a test project and apply the plugin
        Project project = ProjectBuilder.builder().build();
        project.getPlugins().apply("com.example.plugin.greeting");

        // Verify the result
        assertNotNull(project.getTasks().findByName("greet"));
    }
}

File: platforms/documentation/docs/src/snippets/reference/plugin-development/plugin-binary/kotlin/plugin/src/test/kotlin/org/example/GreetingPluginTest.kt
package org.example

import org.gradle.api.Project
import org.gradle.testfixtures.ProjectBuilder
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotNull

class GreetingPluginTest {

    @Test
    fun `plugin registers greet task`() {
        // Create an in-memory project and apply the plugin
        val project: Project = ProjectBuilder.builder().build()
        project.plugins.apply("org.example.greeting")

        // Verify the task is registered
        assertNotNull(project.tasks.findByName("greet"))
    }

    @Test
    fun `extension has sensible defaults`() {
        val project: Project = ProjectBuilder.builder().build()
        project.plugins.apply("org.example.greeting")

        val ext = project.extensions.getByType(GreetingExtension::class.java)

        // Defaults from the plugin:
        assertEquals("Hello from plugin", ext.message.get())
        // Default output file is build/greeting.txt
        val defaultFile = ext.outputFile.get().asFile
        assertEquals(project.layout.buildDirectory.file("greeting.txt").get().asFile, defaultFile)
    }
}
